### PR TITLE
Wg/fix schema description default

### PR DESCRIPTION
**Pull request by DaedalusG**
Drift from the addition of a default value to a table schema provides an invalid sql query to fix

Screenshot 2023-06-16 at 1 53 19 PM

```sql
sourcegraph=# ALTER TABLE notebooks ALTER COLUMN public SET DEFAULT ;
ERROR:  syntax error at or near ";"
LINE 1: ALTER TABLE notebooks ALTER COLUMN public SET DEFAULT ;
                                                              ^
```

## Test plan

Tested locally against the example above and against a more complex case.

Introducing the following drift:
```
ALTER TABLE users 
ALTER COLUMN created_at 
SET DEFAULT '2020-01-01 00:00:00';
```

Case DEFAULT should be null:
Screenshot 2023-06-16 at 2 12 14 PM

Case DEFAULT should have a different value:
Screenshot 2023-06-16 at 2 16 52 PM

### CODE REVIEW

File: internal/database/migration/schemas/description.go [hunk 1]
```diff
@@ -253,6 +253,7 @@ func (d ColumnDescription) AlterToTarget(table TableDescription, target ColumnDe
 	}
 	if d.Default != target.Default {
 		statements = append(statements, fmt.Sprintf("ALTER TABLE %s ALTER COLUMN %s SET DEFAULT %s;", table.Name, target.Name, target.Default))
+		// TODO: handle for DROP DEFAULT rather than
```
Comment (efritz) [inline internal/database/migration/schemas/description.go:256]: TODO :trollface:

Comment (DaedalusG) [reply internal/database/migration/schemas/description.go]: :trollface:

File: internal/database/migration/schemas/description.go [hunk 2]
```diff
@@ -252,7 +252,14 @@ func (d ColumnDescription) AlterToTarget(table TableDescription, target ColumnDe
...
 	if d.Default != target.Default {
-		statements = append(statements, fmt.Sprintf("ALTER TABLE %s ALTER COLUMN %s SET DEFAULT %s;", table.Name, target.Name, target.Default))
+		var verb string
+		if target.Default == "" {
+			verb = "DROP"
+			statements = append(statements, fmt.Sprintf("ALTER TABLE %s ALTER COLUMN %s %s DEFAULT;", table.Name, target.Name, verb))
+		} else {
+			verb = "SET"
+			statements = append(statements, fmt.Sprintf("ALTER TABLE %s ALTER COLUMN %s %s DEFAULT %s;", table.Name, target.Name, verb, target.Default))
+		}
```
Comment (efritz) [inline internal/database/migration/schemas/description.go:262]:
```suggestion
		if target.Default == "" {
			statements = append(statements, fmt.Sprintf("ALTER TABLE %s ALTER COLUMN %s DROP DEFAULT;", table.Name, target.Name))
		} else {
			statements = append(statements, fmt.Sprintf("ALTER TABLE %s ALTER COLUMN %s SET DEFAULT %s;", table.Name, target.Name, target.Default))
		}
```

You don't need to declare the outer-scope variable as you only assign it in the if consequent that you use it in.

**Review by efritz — APPROVED** [review]
i wrote this lol

**Pull request merged**